src/hooks/useDatePicker.ts

```diff
@@ -0,0 +1,247 @@
+import { Temporal } from "@js-temporal/polyfill";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { numberingSystems } from "../constants";
+import {
+  customCalendars,
+  getCustomCalendarLocale,
+  getCustomCalendarLocales,
+} from "../custom-calendars";
+import { SupportedCalendar } from "../types";
+import { isCustomCalendar } from "../utils/helpers";
+import { useCalendarWeekDays } from "./internal/useCalendarWeekDays";
+import {
+  useNavigation,
+  UseNavigationReturnType,
+} from "./internal/useNavigation";
+import { useWeekDayLabels } from "./internal/useWeekDayLabels";
+import "../date-override";
+
+type DatePickerOptions = {
+  date: string;
+  options: LocaleOptions;
+  onDateSelect: ({
```

Comment:
I just wonder if it will cause more problems for downstream users of the Calendar who might expect a JavaScript date and then will try to do something like `getFullYear()` ?